lua/nvim-tree.lua

```diff
@@ -258,13 +251,7 @@ function M.on_enter(netrw_disabled)
     end
   end
 
-  -- Session that left a NvimTree Buffer opened, reopen with it
-  local existing_tree_wins = find_existing_windows()
-  if existing_tree_wins[1] then
-    api.nvim_set_current_win(existing_tree_wins[1])
-  end
-
-  if should_open or existing_tree_wins[1] ~= nil then
```

Comment:
i think this is needed for user using sessions

Reply:
maybe a better fix would be to change the matching string to `NvimTree_`

Reply:
Good catch; I didn't think of nvim sessions, just user calling setup multiple times.

Added a util to check that the buffer name matches and that it is a real file. We can't test file type as it's not imported via default `sessionoptions`.

@barrett-ruth I'd be grateful for a re-test.